packages/dialtone-vue2/components/pagination/pagination.vue

```diff
@@ -181,10 +191,16 @@ export default {
       const end = this.totalPages - start + 1;
 
       if (this.currentPage < start) {
+        if (this.hideEdges) {
+          return [...this.range(1, start + 1), '...'];
+        }
         return [...this.range(1, start), '...', this.totalPages];
       }
 
       if (this.currentPage > end) {
+        if (this.hideEdges) {
+          return ['...', ...this.range(end - 1, this.totalPages)];
+        }
         return [1, '...', ...this.range(end, this.totalPages)];
```

Comment:
Build the array instead of returning two nearly identical cases

```
pages = ['...', ...this.range(end, this.totalPages)];
if (!this.hideEdges) {
  pages.unshift(1);
}
return pages;
```